airbyte-db/src/main/java/io/airbyte/db/bigquery/BigQueryUtils.java

```diff
@@ -90,10 +90,26 @@ private static void setJsonField(Field field, FieldValue fieldValue, ObjectNode
     } else if (fieldValue.getAttribute().equals(Attribute.REPEATED)) {
       ArrayNode arrayNode = node.putArray(fieldName);
       StandardSQLTypeName fieldType = field.getType().getStandardType();
-      fieldValue.getRepeatedValue().forEach(arrayFieldValue -> fillObjectNode(fieldName, fieldType, arrayFieldValue, arrayNode.addObject()));
+      FieldList subFields = field.getSubFields();
+      // Array of primitive
+      if (subFields == null || subFields.isEmpty()) {
```

Comment:
@DoNotPanicUA new  block of code looks too complex. Can you please refactor it a bit within a stream api, etc?